pkg/validation/internal/object.go

```diff
@@ -0,0 +1,198 @@
+func validatePriorityClass(u *unstructured.Unstructured) (result errors.ManifestResult) {
+	pc := schedulingv1.PriorityClass{}
+
+	b, err := u.MarshalJSON()
+	if err != nil {
+		result.Add(errors.ErrInvalidParse("error converting unstructured", err))
+		return
+	}
+
+	err = json.Unmarshal(b, &pc)
+	if err != nil {
+		result.Add(errors.ErrInvalidParse("error unmarshaling priorityclass", err))
+		return
+	}
+
+	if pc.GlobalDefault {
+		result.Add(errors.ErrInvalidObject(pc, "globalDefault field cannot be set to true"))
+	}
+
+	return
+}
+
+func validateRBAC(u *unstructured.Unstructured) (result errors.ManifestResult) {
+	var policyRules []rbacv1.PolicyRule
+
+	b, err := u.MarshalJSON()
+	if err != nil {
+		result.Add(errors.ErrInvalidParse("error converting unstructured", err))
+		return
+	}
+
+	switch u.GroupVersionKind().Kind {
+	case RoleKind:
+		role := rbacv1.Role{}
+		err = json.Unmarshal(b, &role)
+		if err != nil {
+			result.Add(errors.ErrInvalidParse("error unmarshaling role", err))
+			return
+		}
+		policyRules = role.Rules
+	case ClusterRoleKind:
+		clusterrole := rbacv1.ClusterRole{}
+		err = json.Unmarshal(b, &clusterrole)
+		if err != nil {
+			result.Add(errors.ErrInvalidParse("error unmarshaling clusterrole", err))
+			return
+		}
+		policyRules = clusterrole.Rules
+	}
+
+	return audit(policyRules)
+}
+
+// audit checks the provided rbac policies against prescribed limitations.
+// If permission is granted to create/modify a PDB, a warning is returned.
+// If permission is granted to modify default SCCs in OpenShift, an error is returned.
+func audit(policies []rbacv1.PolicyRule) (result errors.ManifestResult) {
+	// check for permission to modify/create PDBs
+	for _, rule := range policies {
+		if contains(rule.APIGroups, PodDisruptionBudgetAPIGroup) &&
+			contains(rule.Resources, "poddisruptionbudgets") &&
+			contains(rule.Verbs, rbacv1.VerbAll, "create", "update", "patch") {
+			result.Add(errors.WarnInvalidObject("RBAC includes permission to modify/create poddisruptionbudgets", rule))
+		}
+	}
+
+	// check sccs for modifying default known SCCs
+	for _, rule := range policies {
+		if contains(rule.APIGroups, SCCAPIGroup) &&
+			contains(rule.Resources, "securitycontextconstraints") &&
+			containsDefaults(rule.ResourceNames, defaultSCCs) &&
+			contains(rule.Verbs, rbacv1.VerbAll, "delete", "update", "patch") {
+			result.Add(errors.ErrInvalidObject(rule, "RBAC includes permission to modify default SCCs"))
+		}
+	}
+
+	return
+}
+
+// contains returns true if at least one item is present in the array
+func contains(slice []string, items ...string) bool {
+	set := make(map[string]struct{}, len(slice))
+	for _, s := range slice {
+		set[s] = struct{}{}
+	}
+
+	for _, item := range items {
+		if _, ok := set[item]; ok {
+			return true
+		}
+	}
+
+	return false
+}
+
+func containsDefaults(slice []string, defaults map[string]struct{}) bool {
+	for key := range defaults {
```

Comment:
Would it be better to flip this around so that it iterates over `slice` once time and checks each element against the map? I assume the inputs are always really small to this, but the growth rate is still bigger than it needs to be.